core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java

```diff
@@ -749,6 +750,23 @@ private static List<IndicatorQueryTreatment> getIndicatorQueryTreatments(Set<Evi
                                     indicatorQueryTreatment.setLevelExcludedCancerTypes(excludedCancerTypes);
                                 }
                                 indicatorQueryTreatment.setDescription(SummaryUtils.enrichDescription(descriptionMap.get(treatment), queryHugoSymbol));
+
+                                ClinicalTrialsUtils clinicalTrialUtils = new ClinicalTrialsUtils();
+                                List<Trial> trials = new ArrayList<>();
+                                String oncotreeCode = tumorType.getCode();
+                                for (Drug drug : treatment.getDrugs()) {
+                                    String drugName = drug.getDrugName();
+                                    try {
+                                        List<Trial> trialsByTreatmentAndCode = clinicalTrialUtils.trialsMatchingGet(oncotreeCode,drugName);
+                                        trials.addAll(trialsByTreatmentAndCode);
+                                    } catch (Exception e) {
+                                        System.out.println(e.getMessage());
+                                        System.out.println("Trials could not be added successfully to IndicatorQueryTreatment.");
+                                    }
+                                }
+                                Set<Trial> trialsSet = new HashSet<>(trials);
+                                indicatorQueryTreatment.setTrials(trialsSet);
```

Comment:
We need to intersection of trials based on individual drug name. Trials that include all drugs should be returned.

Reply:
@zhx828 can you clarify this? Here I am adding all of the trials that contain a drug name from the treatment list. Is that not what you are asking for?